packages/grafana-e2e/cypress/plugins/index.js

```diff
@@ -1,6 +1,13 @@
 const fs = require('fs');
 const path = require('path');
 
+const packageJson = require('../../package.json');
+
+// Only run this plugin if the version of Cypress is 9.x.x
+if (!packageJson?.dependencies?.cypress?.match(/\b9\b\./)) {
```

Comment:
Only run in Cypress 9 environments

Reply:
when does this ever run? to me it looks like it's grabbing the `package.json` for `@grafana/e2e` which has the dep set to 12 now? 🤔

Reply:
that's true in the context of grafana. Plugins on the other hand define and resolve their own versions of grafana packages such as e2e and cypress. So when a plugin runs cypress 9 with the newest e2e package, this will jump into action.

Reply:
got it 👍 this is the bit i'm least sure about tbh because i don't really know the install flow for a plugin. can it ever install `@grafana/e2e` via yarn pnp? cause then this path resolution definitely won't work 😅

Reply:
For a lot of plugins cypress is a nested dependency of the e2e package, they don't declare it. Also create-plugin templates don't include cypress as a dependency. I'm not quire sure what the solution  is (I need to familiarise myself more with the changes in this PR) but I don't think this will work as the path will always resolve to the e2e package.json which will be cypress 12. I also don't think we can assume the location of packages or package.json files either regardless of package manager as the plugin might be in a nested folder or a mono-repo for all we know.